prboom2/src/p_pspr.c

```diff
@@ -257,6 +306,25 @@ int P_WeaponPreferred(int w1, int w2)
    ))))))))))))))));
 }
 
+//
+// P_GetAmmoPerShot
+// Returns ammo per shot for the currently selected weapon.
+//
+// [XA] Eventually this value will be customizable
+// instead of hardcoded, presuming everything goes
+// according to plan...
+//
+
+int P_GetAmmoPerShot(player_t *player)
```

Comment:
This isn't hit by any existing code, right? It will only be hit downstream of the new code pointers? Just making sure heretic doesn't need to be considered :^)

Reply:
Currently no, but the eventual plan is for P_CheckAmmo to call this function `if (mbf21)` in my eventual `ammopershot` PR. Means it still ought not affect Heretic though; presuming you can't set the new complevel there. ;)

Reply:
Instead of a method, what do you think about using an array `shotammo` to match the existing `maxammo` and `clipammo`? That will fit in with the use of `maxammo` and `clipammo` in d_deh, assuming we will be making it all configurable the same way.

Reply:
That makes sense -- though that seems like a thing to do for the new `ammopershot` PR. I'll probably rip out the P_GetAmmoPerShot function completely if it's indeed as simple as indexing an array. ;)